### PR TITLE
[CSL-1906] Update build/cardano-sl.sh script

### PR DESCRIPTION
**Pull request by gromakovsky**
1. Updated the description, it was outdated.
2. Removed obsolete unused stuff (prodMode, bench_mode variables and similar).
3. Removed the logic to treat packages dependent on cardano-sl differently. It
was fairly old logic, which is likely not needed anymore, but leads to
boilerplate and copy-paste when we add a new package.

### CODE REVIEW

File: scripts/build/cardano-sl.sh [hunk 1]
```diff
@@ -191,32 +160,33 @@ xgrep="((^.*warning.*$|^.*error.*$|^    .*$|^.*can't find source.*$|^Module impo
...
-  for prj in $projects; do cleanPackage "cardano-sl-$prj"; done
   exit
 fi
 
 to_build=''
 if [[ $spec_prj == "" ]]; then
   for prj in $projects; do
-    to_build="$to_build cardano-sl-$prj"
-  done
+    if [[ $prj == "lib" ]]; then
```
Comment (volhovm) [inline scripts/build/cardano-sl.sh]: Copypaste. I'll generalize this in the branch, please don't force-push.

File: scripts/build/cardano-sl.sh [hunk 2]
```diff
@@ -191,32 +160,33 @@ xgrep="((^.*warning.*$|^.*error.*$|^    .*$|^.*can't find source.*$|^Module impo
...
+      pkgName="cardano-sl"
+    else
+      pkgName="cardano-sl-$prj"
+    fi
 
-  to_build="$to_build cardano-sl cardano-sl-auxx cardano-sl-tools cardano-sl-wallet cardano-sl-wallet-new cardano-sl-explorer cardano-sl-node"
+    to_build="$to_build $pkgName"
+  done
 
 elif [[ $spec_prj == "lib" ]]; then
```
Comment (volhovm) [inline scripts/build/cardano-sl.sh]: This part can be simplified also.

**Comment by volhovm** [comment]
@gromakovsky please check my changes also.

**Comment by gromakovsky** [comment]
K P A C U B O

**Pull request merged**